### ISSUE TITLE
Support PEP 232 – Function Attributes

### ISSUE THREAD

**Issue by FabriceSalvaire** [astral-sh/ty#4517]
### Summary

It looks like [PEP 232 – Function Attributes](https://peps.python.org/pep-0232) is not supported.

Look at this [Playground](https://play.ty.dev/354b3517-cb91-4378-9656-3a5ce7efcfd5):

```python
def func() -> None:
    func.foo: int  # ruff: ignore[unintentional-type-annotation]
    print('func is running', func.foo)  # ty: ignore[unresolved-attribute]

func.foo = 123  # ty: ignore[unresolved-attribute]
print('This is a valid func attribute', func.foo)  # ty: ignore[unresolved-attribute]
func()
```

Notice, the use cases of PEP 232 – Function Attributes is not so clear for me...

See also #1889

**Comment by AlexWaygood** [comment]
Thanks! Please see #1256

**Comment by sharkdp** [comment]
PEP 232 far predates typing in Python. Allowing arbitrary attribute writes and reads is disallowed (from a type checker perspective) for the same reason that it's disallowed to write/read arbitrary attributes on classes. See https://github.com/astral-sh/ty/issues/1256 for a prior discussion of this topic (and how you can work around this, if you really want, using intersection types)
